### ISSUE TITLE
Null reference on export default anonymous function

### ISSUE THREAD

**Issue by marcind** [brigand/babel-plugin-flow-react-proptypes#146]
Flow: 0.59.0
babel-plugin-flow-react-proptypes: 9.1.0

I have an HOC that looks like this
```javascript
export default function<WrappedProps: {}> (
  WrappedComponent: React.ComponentType<{ injectedProp: string } & WrappedProps>,
): React.ComponentType<WrappedProps> {
   ...
}
```
and causes a `Cannot read property 'name' of null` on this line: https://github.com/brigand/babel-plugin-flow-react-proptypes/blob/0bb901a8b21f759b941540e2d6ae644c3aec52f7/src/index.js#L89

Some quick monkey-patched logging indicates that for the above code snippet, the path.type is a [FunctionDeclaration](https://github.com/babel/babylon/blob/master/ast/spec.md#functiondeclaration). The documentation has this to say:

> Note that unlike in the parent interface Function, the id cannot be null, except when this is the child of an ExportDefaultDeclaration.

Changing the snippet to use a named function:
```javascript
export default function Wrapper<WrappedProps: {}>(
  WrappedComponent: React.ComponentType<{ injectedProp: string } & WrappedProps>,
): React.ComponentType<WrappedProps> {
   ...
}
```
works around the null ref bug.

**Comment by brigand** [comment]
@marcind with the way the code works currently, we can't really support this. It could either silently fail, or give a better error message. Not sure which is least bad.

**Comment by marcind** [comment]
> with the way the code works currently, we can't really support this

I know next to nothing about how Babel works, but is the problem that with an anonymous function there's nothing to attach the `propTypes` property to? What about transforming the tree into something like

```javascript
export default (function decoratePropTypes(func) {
  func.propTypes = {};
  return func;
})(  original_exported_function_declaration_here  );
```

>  It could either silently fail, or give a better error message. Not sure which is least bad.

Given how an error is already thrown, I'd vote for an error message that actually means something and suggest how to work around the problem (i.e. use a named function or don't use default exports)

**Comment by brigand** [comment]
Yeah, we append a statement that attaches propTypes. Doing it inline is a mess with function declarations and class declarations, which would have to become expressions. We could do it smarter, but I don't have a lot of time to work on this currently. I'll add a better error message for now. Pull requests welcome.

**Comment by brigand** [comment]
Improved the error message. Published as v9.1.1

**Comment by marcind** [comment]
👍 +1 works great, would use again.

Including the message "babel-plugin-flow-react-proptypes attempted to add propTypes to a function/class with no name" for future searchability if anyone encounters this error. The solution is to export a default _named_ (as opposed to anonymous) function/class
